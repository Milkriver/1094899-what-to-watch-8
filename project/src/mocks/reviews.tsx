const reviews = [
  {
    id: 1,
    movieId: 1,
    author: 'Kate Muir',
    date: '2016-12-24',
    rating: 8.9,
    text: 'Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director{&apos;}s funniest and most exquisitely designed films in years.',
  },
  {
    id: 2,
    movieId: 1,
    author: 'Bill Goodykoontz',
    date: '2015-11-18',
    rating: 8.0,
    text: 'Anderson&apos;s films are too precious for some, but for those of us willing to lose ourselves in them, they&apos;re a delight. &quot;The Grand Budapest Hotel&quot; is no different, except that he has added a hint of gravitas to the mix, improving the recipe.',
  },
  {
    id: 3,
    movieId: 1,
    author: 'Amanda Greever',
    date: '2015-11-18',
    rating: 8.0,
    text: 'I didn&apos;t find it amusing, and while I can appreciate the creativity, it&apos;s an hour and 40 minutes I wish I could take back.',
  },
  {
    id: 4,
    movieId: 1,
    author: 'Matthew Lickona',
    date: '2016-12-20',
    rating: 7.2,
    text: 'The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.',
  },
  {
    id: 5,
    movieId: 1,
    author: 'Paula Fleri-Soler',
    date: '2016-12-20',
    rating: 7.6,
    text: 'It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.',
  },
  {
    id: 6,
    movieId: 1,
    author: 'Paula Fleri-Soler',
    date: '2016-12-20',
    rating: 7.0,
    text: 'It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.',
  },
];

export default reviews;
