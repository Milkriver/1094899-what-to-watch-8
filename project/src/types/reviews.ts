export interface IReview {
    id: number,
    movieId: number,
    author: string,
    date: string,
    rating: number,
    text: string,
  }
