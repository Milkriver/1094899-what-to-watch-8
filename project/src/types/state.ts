import { AuthorizationStatus } from '../const';
import { IMovie } from './common';
import { IReviewResponse } from './reviews';

export interface IState {
  currentGenre: string
  movies: IMovie[]
  isDataLoaded: boolean
  authorizationStatus: AuthorizationStatus,
  movie: IMovie,
  sameMovies: IMovie[],
  reviews: IReviewResponse[],
  promoMovie: IMovie,
}
