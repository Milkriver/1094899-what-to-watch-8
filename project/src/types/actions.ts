import { AxiosInstance } from 'axios';
import { Action } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { IState } from './state';

export enum ActionType {
  ChangeGenre = 'movieList/ChangeGenre',
  LoadMovies = 'movieList/LoadMovies',
  LoadSingleMovie = 'movieCard/LoadSingleMovie',
  LoadActiveMovie = 'movieCard/LoadActiveMovie',
  LoadSameGenreMovies = 'movieCard/LoadSameGenreMovies',
  LoadPromoMovie = 'movieCard/LoadPromoMovie',
  LoadFavouriteMovies = 'userList/LoadFavouriteMovies',
  AddFavoriteMovie = 'userList/AddFavoriteMovie',
  RemoveFavoriteMovie = 'userList/RemoveFavoriteMovie',
  LoadReviews = 'movieCard/LoadReviews',
  AddReview = 'movieCard/AddReview',
  RequireAuthorization = 'user/requireAuthorization',
  RequireLogout = 'user/requireLogout',
  LoadUserData = 'user/loadUserData',
}

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, IState, AxiosInstance, Action>;

export type ThunkAppDispatch = ThunkDispatch<IState, AxiosInstance, Action>;
