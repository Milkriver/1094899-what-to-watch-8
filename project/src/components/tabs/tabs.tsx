import { useState } from 'react';
import { IMovie } from '../../types/common';
import { IReviewResponse } from '../../types/reviews';
import { MoviePageDetails } from '../movie-page-details/movie-page-details';
import { MoviePageOverview } from '../movie-page-overview/movie-page-overview';
import { MoviePageReviews } from '../movie-page-reviews/movie-page-reviews';
interface IProps {
  movie: IMovie,
  reviews: IReviewResponse[],
}

export function Tabs({ movie, reviews }: IProps): JSX.Element {
  const [activeTab, setActiveTab] = useState('Overview');
  const handleOverview = () => {
    setActiveTab('Overview');
  };
  const handleDetails = () => {
    setActiveTab('Details');
  };

  const handleReviews = () => {
    setActiveTab('Reviews');
  };

  const changeActiveTab = () => {
    switch (activeTab) {
      case 'Overview':
        return <MoviePageOverview movie={movie} />;
      case 'Details':
        return <MoviePageDetails movie={movie} />;
      case 'Reviews':
        return <MoviePageReviews reviews={reviews} />;
      default:
        return <MoviePageOverview movie={movie} />;
    }
  };

  return (
    <div className="film-card__desc">
      <nav className="film-nav film-card__nav">
        <ul className="film-nav__list">
          <li className={activeTab === 'Overview' ? 'film-nav__item film-nav__item--active' : 'film-nav__item'} onClick={handleOverview}>
            <div className="film-nav__link">Overview</div>
          </li>
          <li className={activeTab === 'Details' ? 'film-nav__item film-nav__item--active' : 'film-nav__item'} onClick={handleDetails}>
            <div className="film-nav__link">Details</div>
          </li>
          <li className={activeTab === 'Reviews' ? 'film-nav__item film-nav__item--active' : 'film-nav__item'} onClick={handleReviews}>
            <div className="film-nav__link" >Reviews</div>
          </li>
        </ul >

      </nav>
      <div>
        {changeActiveTab()}
      </div>
    </div >
  );

}
