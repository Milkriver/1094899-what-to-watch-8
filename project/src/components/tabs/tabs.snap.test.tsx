import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IMovie } from '../../types/common';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Tabs } from './tabs';
import { IReviewResponse } from '../../types/reviews';

const mockMovie: IMovie = {
  id: 1, name: 'name1', posterImage: 'posterImage1', previewImage: 'previewImage1', backgroundImage: 'backgroundImage1',
  backgroundColor: 'black', videoLink: 'videoLink1', previewVideoLink: 'previewVideoLink1',
  description: 'description1', rating: 1, scoresCount: 1, director: 'director1',
  starring: ['actor1', 'actor2'], runTime: 1, genre: 'Comedy', released: 1, isFavorite: true,
};
const mockReviews: IReviewResponse[] = [{ id: 1, user: { id: 1, name: 'reviewer' }, rating: 5, comment: 'cool', date: '2021-01-01T23:59:59' }];

describe('Component: Tabs', () => {

  const mockStore = configureMockStore();
  const store = mockStore({
    MOVIES: {},
    USER: {},
    REVIEWS: {},
  });

  store.dispatch = jest.fn();

  test('should render correctly', () => {
    const page = (
      <Provider store={store}>
        <Tabs
          movie={mockMovie}
          reviews={mockReviews}
        />
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
