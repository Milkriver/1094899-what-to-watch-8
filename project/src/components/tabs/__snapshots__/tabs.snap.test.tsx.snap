// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tabs should render correctly 1`] = `
<div>
  <div
    class="film-card__desc"
  >
    <nav
      class="film-nav film-card__nav"
    >
      <ul
        class="film-nav__list"
      >
        <li
          class="film-nav__item film-nav__item--active"
        >
          <div
            class="film-nav__link"
          >
            Overview
          </div>
        </li>
        <li
          class="film-nav__item"
        >
          <div
            class="film-nav__link"
          >
            Details
          </div>
        </li>
        <li
          class="film-nav__item"
        >
          <div
            class="film-nav__link"
          >
            Reviews
          </div>
        </li>
      </ul>
    </nav>
    <div>
      <div
        class="film-rating"
      >
        <div
          class="film-rating__score"
        >
          1
        </div>
        <p
          class="film-rating__meta"
        >
          <span
            class="film-rating__level"
          >
            Bad
          </span>
          <span
            class="film-rating__count"
          >
            1
          </span>
        </p>
      </div>
      <div
        class="film-card__text"
      >
        <p>
          description1
        </p>
        <p
          class="film-card__director"
        >
          <strong>
            Director: 
            director1
          </strong>
        </p>
        <p
          class="film-card__starring"
        >
          <strong>
            Starring: 
            actor1
            actor2
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
